code/modules/clothing/clothing.dm

```diff
@@ -217,6 +217,41 @@
 		for(var/A in armor)
 			armor[A] -= rand(armor[A]/3, armor[A])
 
+// Redefines the attack proc so that when we attack somebody on fire on help intent with a piece of clothing, we have a small chance of putting out the fire.
+/obj/item/clothing/attack(mob/living/M as mob, mob/living/user as mob, def_zone, var/originator = null)
+	if(iscarbon(user)) // First, we check if it's a a carbon.
+		if(user.a_intent == I_HELP) // Then we check if he wants to help.
+			if(istype(src, /obj/item/clothing/under) || (istype(src, /obj/item/clothing/suit) && !istype(src, /obj/item/clothing/suit/space))) // Then we check if the item is either a jumpsuit or suit-but-not-a-spacesut
+				if(ishuman(M) && M.on_fire) // If he is on fire
+					/var/mob/living/carbon/human/target = new/mob/living/carbon/human()
```

Comment:
no need to create a new human at all, just assign M to it directly

Reply:
/var/mob/living/carbon/human/target = M
code\modules\clothing\clothing.dm:226:error: M: undefined var
DM insults me when I do that though.

Reply:
Remove the first `/`, like so: `var/mob/living/carbon/human/target = M`